core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -51,6 +51,8 @@
     !(defined(KOKKOS_ARCH_KEPLER) || defined(KOKKOS_ARCH_MAXWELL50) ||  \
       defined(KOKKOS_ARCH_MAXWELL52))
 #include <cuda_fp16.h>
+#include <iostream>  // istream & ostream for extraction and insertion ops
```

Comment:
No `<iostream>` in any of our headers
```suggestion
#include <iosfwd>  // istream & ostream for extraction and insertion ops
```